diff --git a/pyqtdistutils.py b/pyqtdistutils.py
index 4874e1aa..cf014d4f 100644
--- a/pyqtdistutils.py
+++ b/pyqtdistutils.py
@@ -99,13 +99,16 @@ class build_ext(distutils.command.build_ext.build_ext):
     def _get_sip_exe(self, build_cmd):
         """Get exe for sip. Sources are:
         --sip-exe option, environment, DEF_SIP_BIN, search on path."""
-        return (
+        sip = (
             build_cmd.sip_exe or
             os.environ.get('SIP_EXE') or
             DEF_SIP_BIN or
             find_on_path(
-                ('sip5', 'sip-qt5', 'sip', 'sip5.exe', 'sip.exe'), 'sip')
+                ('sip5', 'sip-qt5', 'sip5.exe', 'sip.exe', 'sip'), 'sip')
         )
+        print('exe:', repr(sip))
+        print('stat:', os.stat(sip))
+        return sip
 
     def _get_sip_inc_dir(self, build_cmd):
         """Get include directory for sip."""
