diff --git a/pyqtdistutils.py b/pyqtdistutils.py
index 4874e1aa..b59941bf 100644
--- a/pyqtdistutils.py
+++ b/pyqtdistutils.py
@@ -44,13 +44,11 @@ def find_on_path(names, mainname):
 
     mainname is the generic name to report
     """
-    path = os.getenv('PATH', os.path.defpath)
-    pathparts = path.split(os.path.pathsep)
+
     for cmd in names:
-        for dirname in pathparts:
-            cmdtry = os.path.join(dirname.strip('"'), cmd)
-            if os.path.isfile(cmdtry) and os.access(cmdtry, os.X_OK):
-                return cmdtry
+        exe = PyQt5.QtCore.QStandardPaths.findExecutable(cmd)
+        if exe:
+            return exe
     raise RuntimeError('Could not find %s executable' % mainname)
 
 def read_command_output(cmd):
@@ -99,13 +97,14 @@ class build_ext(distutils.command.build_ext.build_ext):
     def _get_sip_exe(self, build_cmd):
         """Get exe for sip. Sources are:
         --sip-exe option, environment, DEF_SIP_BIN, search on path."""
-        return (
+        sip = (
             build_cmd.sip_exe or
             os.environ.get('SIP_EXE') or
             DEF_SIP_BIN or
             find_on_path(
-                ('sip5', 'sip-qt5', 'sip', 'sip5.exe', 'sip.exe'), 'sip')
+                ('sip5', 'sip-qt5', 'sip'), 'sip')
         )
+        return sip
 
     def _get_sip_inc_dir(self, build_cmd):
         """Get include directory for sip."""
